app/main.py

```diff
@@ -1,13 +1,32 @@
+from __future__ import annotations
+
+from typing import Dict, List
+
+
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        "line": error.get("line_number"),
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {
+        "path": file_path,
+        "status": "failed" if errors else "success",
+        "errors": [format_linter_error(error) for error in errors]
+   }
 
 
-def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+def format_linter_report(linter_report: dict) -> dict:
+    return {
+        "path": linter_report.get("path"),
```

Comment:
In this function you should only return previos function in nested list, don't need create and return another dict

Reply:
yes, use `format_single_linter_file` function instead